### DISCUSSION TITLE
Priority features

### DISCUSSION THREAD

**Discussion by OffensiveExile** [TychoTheTaco/Twitch-Drops-Bot#243] [Ideas]
Currently, as it stands right now the tool will only watch games we select in the order in which we prioritise in the config.json file. I would like to see some additions to how the priority system works. One flaw I've noticed is that if a drop pops up and only has an hour window when a game with higher priority and a longer end date, then the user will miss out on that extremely time-limited drop. It would be a great idea and addition to be able to prioritse the drops in a manner that allows the user to collect all drops without fear of missing out. 

For example, if a game has a 4 hour view time drop enabled and the user is eligible for 2 weeks to watch streams to claim the drop, the tool should swap to a drop that has an earlier end date to ensure that all possible drops are claimed. Currently, the user will miss out on the short drops due to poor planning or lack of knowledge that the drop would be available at this time.

Another great addition would be to simply continue watching streams that are not in the priority list as long as the accounts are connected and the user would be able to claim the drop. This idea only came into my mind today as I didn't notice a specific game drop was available and I hadn't already put the game ID in my config even though my accounts are connected due to another game associated with the same game's company.

EDIT: I do see there is an option for the tool to watch streams that are not listed within the games section but that doesn't account for connected accounts.

**Comment by TychoTheTaco** [comment]
It's a bit tricky to do that since just because we have 2 weeks left, doesn't mean there will be any streams online during that time. So the following scenario could happen:

Campaign A has 12 hours left and we still need to watch for 3 hours to claim a drop.
Campaign B is starting soon and will only be active for 6 hours. It requires watching for 4 hours to claim the drop.

In your suggestion we would start watching Campaign A, then switch to campaign B as soon as it is available, watch for 4 hours to claim the drop, and then go back to Campaign A to watch the remaining time. The problem is that the stream for Campaign A might have ended after those 4 hours already (even though the campaign is still active for a few hours). In this case we would miss the drop for Campaign A, which had higher priority on the games list.

That scenario happens often with Rocket League streams, where the campaign will be active for the whole weekend, but there will only be streams during certain hours of the day and we need to watch almost all of it to get all the drops.

So that's why the bot will prioritize the order of the game ID above anything else. Maybe this is something that I could make user configurable via the config.

Good idea about the account connected thing though, I will update it to prioritize those.
